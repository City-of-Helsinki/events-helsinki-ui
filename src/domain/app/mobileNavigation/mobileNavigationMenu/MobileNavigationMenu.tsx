import classNames from "classnames";
import React from "react";
import { useTranslation } from "react-i18next";

import { ReactComponent as FlagInformationIcon } from "../../../../assets/icons/svg/flag-information.svg";
import { ReactComponent as OneEyeSmileyIcon } from "../../../../assets/icons/svg/one-eye-smiley.svg";
import { ReactComponent as RankingStartsRibbonIcon } from "../../../../assets/icons/svg/ranking-stars-ribbon.svg";
import { ReactComponent as TagsSearchIcon } from "../../../../assets/icons/svg/tags-search.svg";
import styles from "./mobileNavigationMenu.module.scss";

interface Props {
  isMenuOpen: boolean;
  onMenuClose: () => void;
}

const NavbarMobile: React.FC<Props> = ({ isMenuOpen, onMenuClose }) => {
  const { t } = useTranslation();

  return (
    <div
      className={classNames(styles.mobileNavigationMenu, {
        [styles.menuOpen]: isMenuOpen
      })}
    >
      <div
        className={classNames(styles.menuItem, styles.searchCategories)}
        onClick={onMenuClose}
      >
        <div className={styles.text}>{t("menu.searchCategories")}</div>
        <div className={styles.iconWrapper}>
          <TagsSearchIcon />
        </div>
      </div>
      <div
        className={classNames(styles.menuItem, styles.searchEvents)}
        onClick={onMenuClose}
      >
        <div className={styles.text}>{t("menu.searchEvents")}</div>
        <div className={styles.iconWrapper}>
          <OneEyeSmileyIcon />
        </div>
      </div>
      <div
        className={classNames(styles.menuItem, styles.mostPopular)}
        onClick={onMenuClose}
      >
        <div className={styles.text}>{t("menu.mostPopular")}</div>
        <div className={styles.iconWrapper}>
          <RankingStartsRibbonIcon />
        </div>
      </div>
      <div
        className={classNames(styles.menuItem, styles.aboutService)}
        onClick={onMenuClose}
      >
        <div className={styles.text}>{t("menu.aboutService")}</div>
        <div className={styles.iconWrapper}>
          <FlagInformationIcon />
        </div>
      </div>
      <div className={styles.login} onClick={onMenuClose}>
        <div className={styles.squareExtension}></div>
      </div>
    </div>
  );
};

export default NavbarMobile;
