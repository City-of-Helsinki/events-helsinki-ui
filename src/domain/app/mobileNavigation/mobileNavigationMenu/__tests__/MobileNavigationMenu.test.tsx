import { shallow } from "enzyme";
import React from "react";
import { I18nextProvider } from "react-i18next";

import i18 from "../../../../../common/translation/i18n/i18nInit";
import MobileNavigationMenu from "../MobileNavigationMenu";

it("MobileNavigationMenu matches snapshot", () => {
  const container = shallow(
    <I18nextProvider i18n={i18}>
      <MobileNavigationMenu isMenuOpen={true} onMenuClose={() => {}} />
    </I18nextProvider>
  );
  expect(container.html()).toMatchSnapshot();
});
