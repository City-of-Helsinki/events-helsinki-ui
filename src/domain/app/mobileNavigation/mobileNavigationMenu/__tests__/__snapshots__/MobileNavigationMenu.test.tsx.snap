// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigationMenu matches snapshot 1`] = `"<div class=\\"mobileNavigationMenu menuOpen\\"><div class=\\"menuItem searchCategories\\"><div class=\\"text\\">Selaa kategorioita</div><div class=\\"iconWrapper\\"><svg>tags-search.svg</svg></div></div><div class=\\"menuItem searchEvents\\"><div class=\\"text\\">Etsi tekemist√§</div><div class=\\"iconWrapper\\"><svg>one-eye-smiley.svg</svg></div></div><div class=\\"menuItem mostPopular\\"><div class=\\"text\\">Suosituimmat haut</div><div class=\\"iconWrapper\\"><svg>ranking-stars-ribbon.svg</svg></div></div><div class=\\"menuItem aboutService\\"><div class=\\"text\\">Tietoa palvelusta</div><div class=\\"iconWrapper\\"><svg>flag-information.svg</svg></div></div><div class=\\"login\\"><div class=\\"squareExtension\\"></div></div></div>"`;
